src/analysis/normed_space/operator_norm.lean

```diff
@@ -0,0 +1,217 @@
+    on L(E,F).
+-/
+import algebra.module
+import analysis.normed_space.bounded_linear_maps
+
+variables {k : Type*} 
+variables {E : Type*} {F : Type*}
+
+
+-- Define the subspace of bounded linear maps, introduce the notation L(E,F) for the set of bounded linear maps.
+section bounded_linear_maps
+
+variables [hnfk : normed_field k] [normed_space k E] [normed_space k F]
+include hnfk
+
+def bounded_linear_maps : subspace k (E → F) :=
+{ carrier := {A : E → F | is_bounded_linear_map A},
+  zero := is_bounded_linear_map.zero,
+  add := assume A B, is_bounded_linear_map.add,
+  smul := assume c A, is_bounded_linear_map.smul c }
+
+local notation `L(` E `,` F `)` := @bounded_linear_maps _ E F _ _ _
+
+/-- Coerce bounded linear maps to functions. -/
+instance bounded_linear_maps.to_fun : has_coe_to_fun $ L(E,F) :=
+{F :=  λ _, (E → F), coe := (λ f, f.val)}
+
+@[extensionality] theorem ext {A B : L(E,F)} (H : ∀ x, A x = B x) : A = B :=
+set_coe.ext $ funext H
+
+/-- Bounded linear maps are ... bounded -/
+lemma exists_bound (A : L(E,F)) : ∃ c, c > 0 ∧ ∀ x : E, ∥A x∥ ≤ c * ∥x∥ := A.property.bound
+
+/-- Bounded linear maps are conveniently bounded on the unit ball. -/
+lemma exists_bound' (A : L(E,F)) : ∃ c, c > 0 ∧ ∀ x : E, ∥x∥ ≤ 1 → ∥A x∥ ≤ c :=
+let ⟨c, _, H⟩ := exists_bound A in
+exists.intro c ⟨‹c > 0›,
+  assume x _,
+  calc ∥A x∥ ≤ c * ∥x∥ : H x
+        ... ≤ c * 1 : (mul_le_mul_left ‹c > 0›).mpr ‹∥x∥ ≤ 1›
+        ... = c : mul_one c⟩
+
+end bounded_linear_maps
+
+/-
+Now define the operator norm. We only do this for normed spaces over ℝ, since we need a
+scalar multiplication with reals to prove that ∥A x∥ ≤ ∥A∥ * ∥x∥. It would be enough to
+have a vector space over a normed field k with a real scalar multiplication and certain
+compatibility conditions.
+
+The main task is to show that the operator norm is definite, homogeneous, and satisfies the
+triangle inequality. This is done after a few preliminary lemmas necessary to deal with cSup.
+-/
+section operator_norm
+
+variables [normed_space ℝ E] [normed_space ℝ F]
+open lattice set
+
+local notation `L(` E `,` F `)` := @bounded_linear_maps _ E F _ _ _
+
+/-- The operator norm of a bounded linear map A : E → F is the sup of
+    the set ∥A x∥ with ∥x∥ ≤ 1. If E = {0} we set ∥A∥ = 0. -/
+noncomputable def operator_norm (A : L(E, F)) : ℝ :=
+Sup (image (λ x, ∥A x∥) {x | ∥x∥ ≤ 1})
+
+noncomputable instance bounded_linear_maps.to_has_norm : has_norm L(E,F) :=
+{norm := operator_norm}
+
+lemma norm_of_unit_ball_bdd_above (A : L(E,F)) : bdd_above (image (norm ∘ A) {x | ∥x∥ ≤ 1}) :=
+let ⟨c, _, H⟩ := (exists_bound' A : ∃ c, c > 0 ∧ ∀ x : E, ∥x∥ ≤ 1 → ∥A x∥ ≤ c) in
+bdd_above.mk c
+  (assume r ⟨x, (_ : ∥x∥ ≤ 1), (_ : ∥A x∥ = r)⟩,
+    show r ≤ c, from 
+      calc r = ∥A x∥ : eq.symm ‹∥A x∥ = r›
+         ... ≤ c : H x ‹∥x∥ ≤ 1›)
+
+lemma zero_in_im_ball (A : L(E,F)) : (0:ℝ) ∈ {r : ℝ | ∃ (x : E), ∥x∥ ≤ 1 ∧ ∥A x∥ = r} :=
+have A 0 = 0, from linear_map.map_zero {to_fun := A.val, ..A.property}, 
```

Comment:
Just saw this: this should be a definition `L(E, F) -> L ->l F`, i.e. `bounded_linear_maps` to `linear_map`.